src/Common/src/System/Security/Cryptography/DerEncoder.cs

```diff
@@ -723,6 +785,109 @@ internal static byte[][] SegmentedEncodeIA5String(char[] chars, int offset, int
             };
         }
 
+        internal static byte[][] SegmentedEncodeUtcTime(DateTime utcTime)
+        {
+            Debug.Assert(utcTime.Kind == DateTimeKind.Utc);
+
+            // Low-allocating encoding of YYMMDDHHmmssZ
```

Comment:
This certainly isn't on the hot path, no.  I was mainly feeling bad about all the allocations performed by this type and figured that I could eliminate the intermediate strings here while also eliminating the need to remember if it's HH or hh, and which of MM and mm is month, and if there were any quirks about YY.